Add documentation about how to use the grafana UI (#1187)
scality/metalk8s: docs/grafana-ui.rst

Grafana UI
==========

.. warning::

>   The following test has been realized on vagrant only - One should list

Reviewer: Should be indented with 3 spaces, not 2

>   all requirements for reaching the UI using a operationnal cluster.

Reviewer: ```suggestion
   all requirements for reaching the UI using a operational cluster.
```

Author: done

You will first need the lastest ``kubectl`` version installed on your host.

> From te bootstrap node, get the port used by Grafana:

Reviewer: `s/te/the`

Author: done

.. code-block:: shell

>   $ kubectl --kubeconfig=/etc/kubernetes/admin.conf get service grafana -n monitoring

Reviewer: ditto spaces

  NAME      TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
  grafana   ClusterIP   10.109.125.193   <none>        3000/TCP   1h

Please note the port used by Grafana (here ``3000``)

To authenticate with the cluster, retrieve the admin kubeconfig on your host:

.. code-block:: shell

>   $ scp root@bootstrap:/etc/kubernetes/admin.conf ./admin.conf

Reviewer: ditto spaces

> Return on you host then forward the port used by Grafana

Reviewer: ```suggestion
Return on your host and forward the port used by Grafana
```

.. code-block:: shell

>   kubectl --namespace monitoring port-forward svc/grafana 3000

Reviewer: ditto spaces

Grafana-UI
==========

.. todo::

>   The following test has been realized on vagrant only - One should list

Reviewer 2: s/vagrant/Vagrant/

Author: correction done.

Reviewer 2: Where?

  all requirements for reaching the UI using a operationnal cluster.

>   On your host, you will need to install kubectl.

Reviewer 3: That shouldn't be part of the `todo` note, IIUC. Instead, write it as a requirement, either in plain text, or in a `.. warning::` directive.

Author: modified. I tried to encompass all the different infrastructure. (I still specified that we use vagrant)

> On your host, connect to the vagran VM

Reviewer 3: Same remark as for #1182 - this documentation isn't targetting Vagrant deployments. We may mention it in some situations, but it shouldn't be as obvious as done here.

Reviewer 3: If I may, here's a suggestion for replacing this procedure:

```
You will first need ``kubectl`` installed on your host.

To authenticate with the cluster, first retrieve the admin kubeconfig on your host:

.. code-block:: shell

   $ scp root@bootstrap:/etc/kubernetes/admin.conf ./admin.conf

Using this file, activate port-forwarding for Grafana on your host:

.. code-block:: shell

   $ kubectl -n monitoring --kubeconfig ./admin.conf port-forward svc/grafana 3000

Now, you can navigate to ``http://localhost:3000`` on your host, which should display Grafana UI.
```

Author: correction applied.

> On your host, connect to the vagran VM

Reviewer 2: missing `t` at `Vagrant` (also capitalize the first `v`).

Author: correction done.

.. code-block:: shell

  $ vagrant ssh

Export your config:

.. code-block:: shell

  $ export KUBECONFIG=/etc/kubenetes/admin.conf

Copy and past the :file:`admin.conf` in :file:`/vagrant`

.. code-block:: shell

  $ cp /etc/kubernetes/admin.conf /vagrant

Return on you host then forward the port used by Grafana

.. code-block:: shell

  kubectl --namespace monitoring port-forward svc/grafana 3000

> Then open your web browser and navigate on ``http://localhost:3000``

Reviewer 2: s/on/to/ ?

Author: correction done.
